From bff6bc4fcf96e02274d3cfbfd082da4f729ba9c3 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Thu, 7 Jan 2021 19:13:38 -0500
Subject: [PATCH 6/6] runtime: lib: Put #include <cxxabi.h> behind
 HAVE_LIBINUWIND.

It's only needed when libunwind is used, and always including it fails
on Windows because the header doesn't exist for MSVC.

Signed-off-by: Ryan Volz <ryan.volz@gmail.com>
---
 gnuradio-runtime/lib/terminate_handler.cc | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/gnuradio-runtime/lib/terminate_handler.cc b/gnuradio-runtime/lib/terminate_handler.cc
index ed3b1d93b..370fcc3b5 100644
--- a/gnuradio-runtime/lib/terminate_handler.cc
+++ b/gnuradio-runtime/lib/terminate_handler.cc
@@ -14,13 +14,12 @@
 #include <regex>
 #include <stdexcept>
 
-#include <cxxabi.h>
-
 #ifdef HAVE_LIBUNWIND
 #define UNW_LOCAL_ONLY
 #include <libunwind.h>
 #include <cstdio>
 #include <cstdlib>
+#include <cxxabi.h>
 #endif
 
 namespace gr {
-- 
2.25.1

